sources/api/apiclient/src/update.rs

```diff
@@ -0,0 +1,393 @@
+command.  So, this is our basic process:
+* Check initial status, saving the timestamp.
+* Issue our request.
+* Poll the status API (with timeout and failure checks) until the timestamp changes.
+* Confirm that the status shows that our requested command was successful.
+
+Note: there's some potential for conflict with other callers, even with the API locking.  If
+someone else issues a command after ours completes, but before we can check, the status could be
+lost.  The update API's state machine ensures we only move in expected directions, and we don't
+want to reinvent its logic here, or be too strict about timing.  We can just time out, or perhaps
+sync up if their request was the same.  If it becomes a problem, we could perhaps use something
+like transactions, or timed lock files, to avoid it.
+*/
+
+use super::{raw_request, raw_request_unchecked};
+use http::StatusCode;
+use log::{debug, info, trace, warn};
+use snafu::{ensure, ResultExt};
+use std::path::Path;
+use std::thread;
+use std::time::Duration;
+
+/// Refresh the list of available updates and return the current status.
+pub async fn check<P>(socket_path: P) -> Result<String>
+where
+    P: AsRef<Path>,
+{
+    info!("Refreshing updates...");
+    let (_body, status) = wait_request(
+        socket_path,
+        "/actions/refresh-updates",
+        "POST",
+        None,
+        "refresh",
+        &WaitPolicy::new(Duration::from_millis(100), 10 * 10),
+    )
+    .await?;
+
+    Ok(status)
+}
+
+/// Checks whether an update is required given the current update status, and informs the user
+/// about the specific state.
+// Note: we shouldn't reimplement the update API's state tracking or knowledge here, but giving
+// a useful message rather than "failed to prepare update" in common scenarios is critical.
+pub fn required(check_output: &str) -> bool {
+    let state =
+        response_field(&["update_state"], check_output).unwrap_or_else(|| "unknown".to_string());
+    match state.as_ref() {
+        "Idle" => {
+            info!("No updates available.");
+            false
+        }
+        "Ready" => {
+            info!("Update already applied; reboot for it to take effect, or request a cancel.");
+            false
+        }
+        _ => {
+            debug!("Saw state '{}', assuming update is required", state);
+            true
+        }
+    }
+}
+
+/// Applies the update shown as selected in the output of check(), and makes it active.
+pub async fn apply<P>(socket_path: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    info!("Downloading and applying update to disk...");
+    let (_body, _status) = wait_request(
+        &socket_path,
+        "/actions/prepare-update",
+        "POST",
+        None,
+        "prepare",
+        &WaitPolicy::new(Duration::from_millis(500), 2 * 60 * 10),
+    )
+    .await
+    .context(error::PrepareUpdate)?;
+
+    info!("Setting the update active so it'll apply on the next reboot...");
```

Comment:
Since these outputs have a way of becoming an informal API:

```suggestion
    info!("Setting the update active so it will apply on the next reboot...");
```